support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/ProductSubscriptionBuilders.scala

```diff
@@ -21,60 +21,13 @@ object ProductSubscriptionBuilders {
       case None => throw new CatalogDataNotFoundException(s"RatePlanId not found for $productDescription")
     }
 
-  def buildContributionSubscription(contribution: Contribution, requestId: UUID, config: BillingPeriod => ZuoraContributionConfig): SubscriptionData = {
-    val contributionConfig = config(contribution.billingPeriod)
-    buildProductSubscription(
-      requestId,
-      contributionConfig.productRatePlanId,
-      List(
-        RatePlanChargeData(
-          ContributionRatePlanCharge(contributionConfig.productRatePlanChargeId, price = contribution.amount) //Pass the amount the user selected into Zuora
-        )
-      ),
-      readerType = Direct
-    )
-  }
-
-  def buildProductSubscription(
```

Comment:
this function moved to the SubscribeItemBuilder, as it needs the `createdRequestId` which is not product-specific state, so isn't available to the individual builders any more.